keyvi/include/keyvi/dictionary/fsa/internal/int_inner_weights_value_store.h

```diff
@@ -38,56 +39,62 @@ namespace dictionary {
 namespace fsa {
 namespace internal {
 
-/**
- * An IntValue store which uses the values as inner weights.
- */
-class IntInnerWeightsValueStore final : public IValueStoreWriter {
+class IntInnerWeightsValueStoreBase {
  public:
   typedef uint32_t value_t;
-  static const uint32_t no_value = 0;
   static const bool inner_weight = true;
 
-  using IValueStoreWriter::IValueStoreWriter;
+  IntInnerWeightsValueStoreBase() {}
 
-  // Because the copy ctor is "user declared", the default ctor is not
-  // generated by the compiler
-  IntInnerWeightsValueStore() = default;
-  IntInnerWeightsValueStore(const IntInnerWeightsValueStore& that) = delete;
-  IntInnerWeightsValueStore& operator=(IntInnerWeightsValueStore const&) = delete;
+  IntInnerWeightsValueStoreBase(const IntInnerWeightsValueStoreBase& that) = delete;
+  IntInnerWeightsValueStoreBase& operator=(IntInnerWeightsValueStoreBase const&) = delete;
 
-  uint64_t GetValue(value_t value, bool* no_minimization) const {
+  uint64_t AddValue(value_t value, bool* no_minimization) const {
     TRACE("Value store value: %d", value);
 
     return value;
   }
-
   uint32_t GetWeightValue(value_t value) const { return value; }
 
-  uint64_t GetMergeValueId(size_t fileIndex, uint64_t oldIndex) { return oldIndex; }
-
   uint32_t GetMergeWeight(uint64_t fsa_value) { return fsa_value; }
 
-  static value_store_t GetValueStoreType() { return INT_INNER_WEIGHTS_VALUE_STORE; }
-
+  void CloseFeeding() {}
   void Write(std::ostream& stream) const {}
 
-  /**
-   * Close the value store, so no more updates;
-   */
-  void CloseFeeding() {}
+  static value_store_t GetValueStoreType() { return value_store_t::INT_WITH_WEIGHTS; }
+};
 
- private:
-  template <typename, typename>
-  friend class ::keyvi::dictionary::DictionaryMerger;
+/**
+ * An IntValue store which uses the values as inner weights.
+ */
+class IntInnerWeightsValueStore final : public IntInnerWeightsValueStoreBase {
```

Comment:
Seems we don't need this class anymore. Maybe it's worth to delete it and just rename `IntInnerWeightsValueStoreBase` -> `IntInnerWeightsValueStore`.

Reply:
With #102  the `AddValue(...)` methods will move out of the base class into here.